src/NuGet.Clients/VisualStudio.Facade/NuGetUIThreadHelper.cs

```diff
@@ -33,28 +33,29 @@ public static JoinableTaskFactory JoinableTaskFactory
         /// Retrieve the CPS enabled JoinableTaskFactory for the current version of Visual Studio.
         /// This overrides the default VsTaskLibraryHelper.ServiceInstance JTF.
         /// </summary>
-        public static void SetJoinableTaskFactoryFromService(IComponentModel componentModel)
+        public static void SetJoinableTaskFactoryFromService(IProjectServiceAccessor projectServiceAccessor)
         {
-            if (componentModel == null)
+            if (projectServiceAccessor == null)
             {
-                throw new ArgumentNullException(nameof(componentModel));
+                throw new ArgumentNullException(nameof(projectServiceAccessor));
             }
 
-            _joinableTaskFactory = new Lazy<JoinableTaskFactory>(() =>
+            if (_joinableTaskFactory == null)
```

Comment:
`_joinableTaskFactory` is a private static. public property to access it is via `JoinableTaskFactory` which already has clear doc comment like below:
`/// Returns the static instance of JoinableTaskFactory set by SetJoinableTaskFactoryFromService.
/// If this has not been set yet the shell JTF will be used.
/// During MEF composition some components will immediately call into the thread helper before
/// it can be initialized. For this reason we need to fall back to the default shell JTF
/// to provide basic threading support.`

And also `SetJoinableTaskFactoryFromService` api has a clear doc comment as below:
`/// Retrieve the CPS enabled JoinableTaskFactory for the current version of Visual Studio.
/// This overrides the default VsTaskLibraryHelper.ServiceInstance JTF.`